### ISSUE TITLE
Manual page for jupyter-server

### ISSUE THREAD

**Issue by SnarkBoojum** [jupyter/jupyter_server#320]
Hi,

I'm currently packaging jupyter_server for Debian, and prepared a manual page for the command line utility. It was done using help2man and elbow grease.

[jupyter-server.1.txt](https://github.com/jupyter/jupyter_server/files/5397149/jupyter-server.1.txt)

**Comment by kevin-bates** [comment]
This is pretty cool.  I copied the file into my mac's man1 directory and can run `man jupyter-server`.  Given this information is already documented I'm not sure if its deployment would be more work than it's worth - given all the potential for platform differences, access to locations, etc.  

What is your intention?  Are man pages typically "write once" or are there variations per platform, etc.?

**Comment by SnarkBoojum** [comment]
I ship it within the Debian package, so I propose it upstream because it's a more natural place to find it.

There are no variations per platform as far as I know : it's a self-contained document.

But of course it can have sections indicating that they make sense only in such and such situations.

**Comment by kevin-bates** [comment]
OK.  I agree that it makes sense to be stored in the repo, but I'm not sure where the best location would be.  In `docs` makes some sense, but if the intention is to "install" the man page during installation, then that's probably not the best location.  I'm also concerned about its maintainability as this will easily get out of sync.

Is there precedent for this in other Jupyter repos?
To be honest, it seems like a `--man-page` option (similar to `--help`) would be the approach to take - which would imply this be done in the [`traitlets` package](https://github.com/ipython/traitlets) and would then essentially enable this functionality across the ecosystem.

**Comment by jasongrout** [comment]
It seems we can distribute it with the python package, like this? https://stackoverflow.com/a/48381612 



> Is there precedent for this in other Jupyter repos?

I don't think we ship manpages with other Jupyter repos. We probably should?



>  To be honest, it seems like a `--man-page` option (similar to `--help`) would be the approach to take - which would imply this be done in the [`traitlets` package](https://github.com/ipython/traitlets) and would then essentially enable this functionality across the ecosystem.

That makes a lot of sense. Do you mean that `--man-page` would generate a manpage, and that generating such a page would be a normal part of the packaging process for Jupyter repos?

**Comment by kevin-bates** [comment]
Thanks for the link Jason.

> Do you mean that --man-page would generate a manpage, 

Yes.

> and that generating such a page would be a normal part of the packaging process for Jupyter repos?

Given your link, it does seem like we'd have a well-known location within the repo for these and the "release script" could invoke the man-page generation as part of the tagging step, etc. to maintain consistency. (I'm not completely up on all the various packaging tricks that apply to setup.py.)

I think the most important point is that there be a way to generate the man-page entry(ies) and not rely on manual updates.

**Comment by jasongrout** [comment]
> I think the most important point is that there be a way to generate the man-page entry(ies) and not rely on manual updates.

+1 to making this maintenance free, especially since none of us have maintained manpages before (or perhaps you have, which is why you are insisting on no manual updates? :).

@SnarkBoojum, are you willing to maintain this manpage if it is merged here?

**Comment by SnarkBoojum** [comment]
The idea to have the manpage automagically produced through a --manpage switch is technically the best solution : I'm casting my vote on it!

**Comment by jasongrout** [comment]
Awesome. We'd need some help, given we don't have experience with writing manpages. I've opened https://github.com/ipython/traitlets/issues/636. If you want to work on it, it would be much appreciated!

**Comment by jasongrout** [comment]
Should we close this issue, then, since we don't have a maintainer for the manpage here, and shift the conversation over to traitlets in hopes that someone can write an automated manpage generator?

**Comment by SnarkBoojum** [comment]
No, it should be closed when jupyter_server will actually provide the manpage. Keep it as a reminder.
